core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -0,0 +1,738 @@
+                    thread_id,
+                    transaction,
+                    max_age,
+                    cost,
+                }) => {
+                    assert!(
+                        self.working_account_set.take_locks(&transaction),
+                        "locks must be available"
+                    );
+                    saturating_add_assign!(num_scheduled, 1);
+                    batches.transactions[thread_id].push(transaction);
+                    batches.ids[thread_id].push(id.id);
+                    batches.max_ages[thread_id].push(max_age);
+                    saturating_add_assign!(batches.total_cus[thread_id], cost);
+
+                    // If target batch size is reached, send all the batches
+                    if batches.ids[thread_id].len() >= self.config.target_transactions_per_batch {
+                        saturating_add_assign!(num_sent, self.send_batches(&mut batches,)?);
+                        self.working_account_set.clear();
+                    }
+
+                    // if the thread is at max_cu_per_thread, remove it from the schedulable threads
+                    // if there are no more schedulable threads, stop scheduling.
+                    if self.in_flight_tracker.cus_in_flight_per_thread()[thread_id]
+                        + batches.total_cus[thread_id]
+                        >= max_cu_per_thread
+                    {
+                        schedulable_threads.remove(thread_id);
+                        if schedulable_threads.is_empty() {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        self.working_account_set.clear();
+        self.send_batches(&mut batches)?;
+
+        // Push unschedulables back into the queue
+        for id in self.unschedulables.drain(..) {
+            container.push_id_into_queue(id);
+        }
+
+        Ok(SchedulingSummary {
+            num_scheduled,
+            num_unschedulable,
+            num_filtered_out,
+            filter_time_us: total_filter_time_us,
+        })
+    }
+
+    /// Receive completed batches of transactions without blocking.
+    /// Returns (num_transactions, num_retryable_transactions) on success.
+    fn receive_completed(
+        &mut self,
+        container: &mut impl StateContainer<Tx>,
+    ) -> Result<(usize, usize), SchedulerError> {
+        let mut total_num_transactions: usize = 0;
+        let mut total_num_retryable: usize = 0;
+        loop {
+            let (num_transactions, num_retryable) = self.try_receive_completed(container)?;
+            if num_transactions == 0 {
+                break;
+            }
+            saturating_add_assign!(total_num_transactions, num_transactions);
+            saturating_add_assign!(total_num_retryable, num_retryable);
+        }
+        Ok((total_num_transactions, total_num_retryable))
+    }
+}
+
+impl<Tx: TransactionWithMeta> GreedyScheduler<Tx> {
```

Comment:
Everything in this impl block is an exact copy from `PrioGraphScheduler`.
Duplicating to avoid refactors in a PR intended to be backported to 2.1.